specifyweb/frontend/js_src/lib/components/SchemaConfig/UniquenessRules.tsx

```diff
@@ -128,22 +132,16 @@ export function TableUniquenessRules({
   );
 
   return (
-    <Dialog
-      buttons={
-        <>
-          <Button.DialogClose>{commonText.close()}</Button.DialogClose>
-          {saveBlocked ? (
-            <Button.Danger
-              className="cursor-not-allowed"
-              onClick={(): void => console.error(tableRules)}
-            >
-              {icons.exclamation}
-              {commonText.save()}
-            </Button.Danger>
-          ) : (
+    <Form id={formId}>
+      <Dialog
+        buttons={
+          <>
+            <Button.DialogClose>{commonText.close()}</Button.DialogClose>
             <Submit.Save
               disabled={tableRules === uniquenessRules[model.name]}
-              onClick={(): void => {
+              form={formId}
+              onClick={(event): void => {
```

Comment:
this is not a correct place for this
can you move this callback function into the Form's `onSubmit` handler?
(and no need for event.preventDefault() as form takes care of that automatically)

(i.e we should be handling form submission, not intercepting user trying to submit the form)

while I am not completely sure, your current implementation may also lead to a bug - the Form's onSubmit is only triggered when form is valid - the submit button's onClick may be triggered even if form contains fields with validation errors

Reply:
oh, looks like you already fixed this